components/sensors/imu/imu.c

```diff
@@ -15,6 +15,11 @@
 #include "fileUtils.h"
 #include "periodicSensor.h"
 
+#ifdef MK_CONFIG_PERIODIC_SENSOR_ABSOLUTE
+#define IMU_PREFIX_NAME   "/app/redSensor/"
+#else
+#define IMU_PREFIX_NAME   ""
```

Comment:
How about each component identifies its "prefix" and the pSensor then chooses whether or not to use that value as the value passed to dhub call?
Actually thinking about it - I wonder if there is an easy way to automatically grab the application name that is being compiled and have the pSensor code use that automatically - instead of each component manually setting it. Let's have a chat about this today.
Might even be able to get rid of all the #if, #else, #endif.

Reply:
Did this conversation take place today?
Several Octave apps already do this.  E.g:
```
const char *orp_MyNameGet
(
    void
)
//--------------------------------------------------------------------------------------------------
{
    static char name[LE_LIMIT_APP_NAME_LEN + 1] = {0};
    static bool init = false;

    if (false == init)
    {
#ifdef LE_CONFIG_RTOS
        strncpy(name, "orp", sizeof(name));
#else
        init = true;
        le_result_t result = le_appInfo_GetName((int32_t)getpid(), name, sizeof(name));
        if (LE_OK != result)
        {
            LE_FATAL("Failed to get application name from pid: %s", LE_RESULT_TXT(result));
        }
#endif
    }
    return (const char *)name;
}
```